### ISSUE TITLE
project builder: auto-generated manifest should be skipped when a manifest is supplied

### ISSUE THREAD

**Issue by vrooje** [zooniverse/Panoptes-Front-End#2241]
I'm uploading a subject set with multiple-image subjects only, and I've made the manifest. With each batch I upload I see a screen like this:
![screen shot 2016-02-29 at 8 47 28 pm](https://cloud.githubusercontent.com/assets/2357126/13419573/0a5ddfe0-df34-11e5-9ee2-808d86b550be.jpg)
I've double-checked with a smaller batch and each of the images in the auto-generated manifest are also in the subjects I've specified in the manifest I did upload, so the auto-generated part isn't triggering on subjects I've forgotten to include in the manifest. Below the part I've screenshotted is a button that says "Upload 767 New Subjects"; the number changes to 292 when I click the x to remove the auto-generated subjects. That seems to mean that if I clicked submit without removing the auto-generated part I'd end up with the subjects I specified plus a bunch of subjects I didn't, all mixed together in the same subject set.

I haven't tested it by clicking submit because this is not a test subject set and I don't want to risk making subjects I don't want and that I can't easily delete - but if my impression is correct, that's a bug, right? If someone does upload a manifest we shouldn't assume they want single-image subjects with no other metadata than the filename. It's good we give project owners the option to delete the auto-generated stuff, but I don't think this is going to be very clear for newer users of PFE.

**Comment by CKrawczyk** [comment]
I have seen this behavior as well.  +1 for not making an auto generated manifest when one is provided.

**Comment by aliburchard** [comment]
That seems to be a new thing -- I've never gotten it before but haven't uploaded subjects in a month or so...

**Comment by eatyourgreens** [comment]
@vrooje @aliburchard is this still a problem, or can we close this?

**Comment by vrooje** [comment]
I haven't tried uploading subjects in a while so I'm not sure - maybe Ali has done so more recently?

**Comment by vrooje** [comment]
Update: this is definitely still an issue. It was one of the biggest anxiety points during my process of creating and uploading subjects for PRN. I used the project builder to upload new subjects, and this option came up every time despite always having a manifest. Due to the urgency in that project, and the fact that I knew there were people sitting at their computers clicking "refresh" and awaiting new subjects to classify, I would sometimes activate a subject set after uploading the first 1 or 2 batches (and spot checking that, of course), and then continue adding subjects to the subject set while the subject set was "live". Luckily I managed to overcome the sleep deprivation enough to *not* miss deleting the auto-generated manifest option during any batch of new subject uploads... but it would be great to not have to rely on my own paranoia, as a mistake will happen eventually.

**Comment by vrooje** [comment]
Yeah, this is still happening and I'm not sure why? I saw a live demo with @trouille and @lcjohnso where the PB did *not* do it, but it still happens to me every time. I wondered if it's because I usually drag the images and the manifest in separately.

**Comment by lcjohnso** [comment]
@vrooje: I think you're right -- I always drag both images and manifest in together so have not had a problem.

**Comment by srallen** [comment]
Yes, they have to be dragged in together. If you drag and drop only subjects, then the code is written to auto-generate a manifest. There is no code written to handle a manifest if it is added later.

**Comment by vrooje** [comment]
If I drag a manifest in later (or first) it figures out the
manifest-generated subjects just fine; it just doesn't figure out that it
should then remove the auto-generated subjects. Is that intended behaviour?
I can't think of a use case where you'd want both but I could be too
limited in my thinking.

**Comment by srallen** [comment]
It's been a while since I've closely looked at the code, but here's what I remember:
- If subject files and a manifest are added at the same time, the manifest is parsed and it uses the filenames in the manifest and compares it to the subject files and validates.
- If only a manifest csv is added, it is parsed and validated which then shows errors in the UI that the subject files are missing
- If subject files are added on their own, the auto-generate manifest code runs which grabs the filenames from the subject and creates a manifest csv to use. In this scenario, if a manifest is later added, there is no code written to know which to use which I think is the scenario that is being described in the issue. The manifest is parsed and validated, but fails because it doesn't know there are already subjects and an auto-generated manifest. I don't know of any auto-generated subjects, I think you're referring to the validation messages in the UI after the manifest csv is parsed?

**Comment by vrooje** [comment]
Whether I add the manifest first and the subjects second or the subjects first and the manifest second, I see this:
![screen shot 2018-03-20 at 9 28 51 am](https://user-images.githubusercontent.com/2357126/37668212-2ebc06fa-2c21-11e8-8613-8ac6ae99acb6.jpg)

If I click the upload button without clicking the X on the auto-generated manifest line, it uploads the images and creates *both* the manifest-generated images and the single-image subjects from the auto-generated manifest. If I do click the X, the 2-image subjects in the manifest I specified upload correctly. This happens even if I added the manifest after adding the subject files on their own. (All of this is before clicking the upload button, obviously.)
